### PR TITLE
chore(deps): bump typescript from 5.2.2 to 5.6.2

### PR DESCRIPTION
**Pull request by karrui**
# Component schema changes
Use `Type.Composite` instead of `Type.Intersect` so Typescript does not complain. Typewise equal since there the union type does not contain the base type.

# TypeScript and Dependency Updates

This PR updates the TypeScript version from 5.2.2 to 5.6.2 across various package.json files in the project. Additionally, it includes the following changes:

- Updates @types/node from ^20.14.5 to ^22.5.5 in relevant packages
- Upgrades @sinclair/typebox from ^0.32.34 to ^0.33.12 in the components package
- Refactors ESLint configurations to use the new flat config format
- Adjusts some type definitions and utility functions for improved type safety
- Updates the generated comment in the Prisma selectableTypes file
- Removes unnecessary dependencies from useCallback hooks in the Storybook preview file

These updates aim to improve type checking, take advantage of new TypeScript features, and enhance overall code quality across the project.

### CODE REVIEW

**Comment by karrui** [comment]
* **#667** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/667?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karrui** [comment]
this update i think cursed @dcshzj hitting the limitations of typescript liow

**Comment by dcshzj** [comment]
> this update i think cursed @dcshzj hitting the limitations of typescript liow

Yeah a bit cancer idk what happened here, did they adjust the limit or something?

**Comment by karrui** [comment]
> > this update i think cursed @dcshzj hitting the limitations of typescript liow
> 
> Yeah a bit cancer idk what happened here, did they adjust the limit or something?

I think so. or just more strict. should try to avoid it or later we stuck permanently on typescript v5.2 lol. hard limited by typebox 😆. At most write a script to add ts-ignore on the offending lines.

File: apps/studio/prisma/generate.cts [hunk 1]
```diff
@@ -37,7 +37,7 @@ const HEADER = [
   ts.addSyntheticLeadingComment(
     KYSELY_IMPORT,
     ts.SyntaxKind.SingleLineCommentTrivia,
-    "THIS FILE IS AUTOMATICALLY GENERATED. DO NOT EDIT IT MANUALLY.",
+    " THIS FILE IS AUTOMATICALLY GENERATED. DO NOT EDIT IT MANUALLY.",
```
Comment (harishv7) [inline apps/studio/prisma/generate.cts:40]: nit: why is there a leading space?

**Review by harishv7 — APPROVED** [review]

File: apps/studio/prisma/generate.cts (hunk 1 above)
Comment (karrui) [reply apps/studio/prisma/generate.cts:40]: so the comment looks nice haha. see https://github.com/opengovsg/isomer/pull/667/files#diff-23e6bf6b54d1b3785fa9d5c8684ad931e5cc7eeab03578e8db8ca75344071b0a

**Comment by dcshzj** [comment]
@karrui can merge this now, the schema change is ok to make now

**Pull request merged**